src/main/java/org/cbioportal/service/impl/ClinicalDataDensityPlotServiceImpl.java

```diff
@@ -33,7 +33,7 @@ public DensityPlotData getDensityPlotData(List<ClinicalData> sampleClinicalData,
         result.setBins(new ArrayList<>());
 
         Map<String, List<ClinicalData>> clinicalDataGroupedBySampleId = sampleClinicalData.stream().
-            collect(Collectors.groupingBy(ClinicalData::getSampleId));
+            collect(Collectors.groupingBy(c -> c.getStudyId() + "_" + c.getSampleId()));
```

Comment:
this may not be necessary, but since we removed the study id from the sample id for clinical data this change will keep the ids same for density plot implementation.